Dockerfile

```diff
@@ -108,7 +103,7 @@ ADD .yarnrc /usr/src/
 
 COPY --from=builder /usr/src/packages ./packages
 
-RUN yarn install --production --frozen-lockfile --ignore-scripts --prefer-offline
+RUN --mount=type=cache,uid=1000,gid=1000,target=/home/node/.yarn YARN_CACHE_FOLDER=/home/node/.yarn yarn install --production --frozen-lockfile --ignore-scripts --prefer-offline
```

Comment:
Won't this re-use the cache from the previous step, which we don't want to do because we don't want the dev dependencies? See https://github.com/yarnpkg/yarn/issues/6373#issuecomment-771261476

I mentioned in #2230 that we have to do a clean install on the second step because of this bug. If we're building from the cache, then this entire second stage of the build can be removed completely ie. revert #2230.

Reply:
it'll have the cache artifacts from the `builder` stage available for this yarn cmd `yarn install --production --frozen-lockfile --ignore-scripts --prefer-offline` via the `--mount=type=cache` so it can re-use any packages in the cache and avoid redownloading. 

However it won't persist the cache data in the underlying image, rather it mounts the cache volume at build time for data re-use.

However based on the yarn bug you outline above then this will be a problem. so it's probably best to remove this 2nd re-use of the cache volume and install afresh.

Reply:
```suggestion
RUN yarn install --production --frozen-lockfile --ignore-scripts --prefer-offline
```

Reply:
Yeah, that change should do it. My understanding of https://github.com/yarnpkg/yarn/issues/6373 is that `yarn install --production` installs everything from the yarn cache, even dev dependencies.